internal/orchestrator/updates.go

```diff
@@ -388,28 +434,34 @@ func (o *Orchestrator) eventNeedsReconciliation(evt *v1types.ConditionUpdateEven
 	})
 
 	lastTime, err := registry.LastContact(evt.ControllerID)
-	if errors.Is(err, nats.ErrKeyNotFound) {
-		return true
-	}
+	if err != nil {
```

Comment:
you don't need this. `nil` is not `nats.ErrKeyNotFound`.

_Edit:_ I'm ambivalent at best on this particular refactor. I don't think it adds anything, but instead just pushes code around in different ways.

Reply:
Are you saying the change is incorrect here?

The point with the change was to group the error checks and return earlier, since that is easier to grok (imo),
as part of that it un-indents the additional else clause.

old
```go
	lastTime, err := registry.LastContact(evt.ControllerID)
	if errors.Is(err, nats.ErrKeyNotFound) {
		return true
	}

	if err == nil {
		since := time.Since(lastTime)
		le.WithFields(logrus.Fields{
			"last.update":    evt.UpdatedAt.String(),
			"worker.checkin": since.String(),
		}).Debug("long running event caught by reconciler")
	} else {
		le.WithError(err).Warn("error checking registry")
	}

	return false
```

new
```go
	lastTime, err := registry.LastContact(evt.ControllerID)
	if err != nil {
		if errors.Is(err, nats.ErrKeyNotFound) {
			return true
		}

		le.WithError(err).Warn("error checking registry")
		return false
	}

	// we don't attempt to clean up, but collect metrics on it
	since := time.Since(lastTime)
	le.WithFields(logrus.Fields{
		"last.update":    evt.UpdatedAt.String(),
		"worker.checkin": since.String(),
	}).Debug("long running event caught by reconciler")

	metrics.ConditionReconcileStale.With(
		prometheus.Labels{"conditionKind": string(evt.Kind)},
	).Inc()

	return false
```

Reply:
I think we're just arguing code aesthetics, and that's the root of the issue. This change doesn't make meaningful changes aside shifting things about, and I don't find the new organization an improvement.

Reply:
Clarity in code layout is an improvement. 
The proposal here to have a single block within which all error cases handled, which to me is clearer. Given that we're maintaining this code together, I don't see why we should not proceed with this change.

Reply:
I 100% do not agree that this is "clearer."